blog/_posts/2021-07-06-distributed-rdataframe.md

```diff
@@ -0,0 +1,105 @@
+achieved through a modular implementation that defines a generic task (representing the RDataFrame computation graph) to
+be executed on data. The input dataset is logically split into many ranges of entries, which will be sent to the
+distributed nodes for processing. Each range will then be paired to the generic task and submitted to the computing
+framework via a specific backend implementation. An added benefit of using RDataFrame is that the distributed
+tasks run C++ computations. This is made possible by PyROOT and cling. Currently the package offers support for running
+the application on a Spark cluster, but the package design will make it possible to add many more backends in the future.
+For example, backends for Dask and AWS Lambda have already been implemented and demonstrated at different conferences
+[[1]](#dasksummit)[[2]](#hips21). They will be made available in future ROOT releases.
+
+Great attention is being put towards ensuring that running the RDataFrame computation graph scales well across
+multiple computing nodes and different backend implementations. This has been shown since the first stages of the
+development of this package with a real use case analysis running on a Spark cluster [[3]](#europar19). More recently, a benchmark
+based on CERN open data has shown promising scaling performance with both Spark and Dask [[4]](#pyhep21):
+
+![Scaling of distributed RDataFrame running the dimuon analysis on 100 times the original data. Dask and Spark backends compared]({{ '/assets/images/distrdf_dimuon_scaling_dask_spark_pyhep2021.png' | relative_url }})
+
+All it takes to steer an existing RDataFrame application to distributed resources is to use the correct RDataFrame
+instance and provide it with the arguments needed to connect to the remote cluster. Let's see an example of connection
+to a Dask cluster:
+
+```python
+from dask.distributed import Client
+import ROOT
+
+# Point RDataFrame calls to the Dask specific RDataFrame
+RDataFrame = ROOT.RDF.Experimental.Distributed.Dask.RDataFrame
+
+# Create the Client object to connect to the Dask cluster
+# See the Dask documentation for all the options available
+client = Client("DASK_SCHEDULER_ADDRESS")
+
+# It still accepts the same constructor arguments as traditional RDataFrame
+# And supports some extra keyword arguments
+df = RDataFrame("mytree", "myfile.root", npartitions = 8, daskclient = client)
+```
+
+In this example the `npartitions` parameter tells the RDataFrame into how many ranges of entries the input dataset should
+be split. Each range will then correspond to a task on a node of the cluster. The `daskclient` parameter receives the
+object needed to connect to the Dask scheduler. All the options are available in the
+[Dask documentation](https://distributed.dask.org/en/latest/client.html). The equivalent object for the Spark framework
+is called [SparkContext](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.SparkContext.html) and in
```

Comment:
I think the two code examples (Spark and Dask) should go one after the other. This way you can explain well what I already pointed out in one of my previous comments: you can choose not to provide any connection details for the scheduler (Spark example above) and then everything will happen locally, but you can also configure RDataFrame to use an existing cluster (and you show an example with Dask). It just seems to me that the programming model description and the examples are scattered for no reason.

Reply:
My way of thinking about the distance between the two snippets is to avoid seeing too much code, too close. I think it would be confusing. I do believe you have a good point about the "you can use defaults vs you can provide more connection details". I can still come with something that is closer to your suggestion and send some screenshots to let you see